crates/evm/src/evm/handler.rs

```diff
@@ -557,9 +558,13 @@ fn calc_diff_size<EXT, DB: Database>(
             let account = &state[addr];
 
             if let Some(code) = account.info.code.as_ref() {
-                // if code is eoa code
-                diff_size += CODE_KEY_SIZE;
-                diff_size += code.len();
+                // Don't charge for account code if it is already in DB.
+                let db_code = db.code_by_hash(account.info.code_hash)?;
```

Comment:
Unfortunately there is no way to optimize this, we have to deser the code anyway. So there is no use in those .exists/ contains_key methods at all...

Reply:
look, selfdestruct is very very rarely used. so it's not a big deal